tests/test_edgeql_ddl.py

```diff
@@ -2839,6 +2839,118 @@ async def test_edgeql_ddl_ptr_set_type_validation(self):
                 ALTER TYPE Foo ALTER LINK l SET TYPE Spam USING (SELECT Spam)
             """)
 
+    async def test_edgeql_ddl_ptr_using_dml_01(self):
+        await self.con.execute(
+            r"""
+            CREATE TYPE Hello;
+            CREATE TYPE World {
+                CREATE LINK hell -> Hello;
+                CREATE MULTI LINK heaven -> Hello;
+            };
+            INSERT World {
+                heaven := {
+                    (INSERT Hello),
+                    (INSERT Hello),
+                    (INSERT Hello),
+                }
+            };
+            INSERT World {
+                hell := (INSERT Hello),
+                heaven := {
+                    (INSERT Hello),
+                    (INSERT Hello),
+                }
+            };
+            """
+        )
+        self.assertEqual(len(await self.con.query("SELECT Hello")), 6)
+
+        await self.con.execute(
+            r"""
+            ALTER TYPE World {
+                ALTER LINK hell SET REQUIRED USING (INSERT Hello);
+                ALTER LINK heaven SET SINGLE USING (INSERT Hello);
```

Comment:
We definitely want a test that inserts in a way that refers to existing data.
I mentioned a workaround for the correlated set thing, but I think also we need to fix the correlated set error in this case, since I think it is an important use case.